extensions/amp-base-carousel/amp-base-carousel.md

```diff
@@ -40,9 +40,204 @@ navigational arrows to go forward or backwards a single item.
+
+#### Example
+
+The example below demonstrates `amp-base-carousel` component in standalone use.
+
+```
+<head>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.css">
+  <script async custom-element="amp-base-carousel" src="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.js"></script>
+  <style>
+    amp-base-carousel, amp-base-carousel > div {
+      aspect-ratio: 4/1;
+    }
+    .red-gradient {
+      background: brown;
+      background: linear-gradient(90deg, brown 50%, lightsalmon 90%, wheat 100%);
+    }
+    .blue-gradient {
+      background: steelblue;
+      background: linear-gradient(90deg, steelblue 50%, plum 90%, lavender 100%);
+    }
+    .green-gradient {
+      background: seagreen;
+      background: linear-gradient(90deg, seagreen 50%, mediumturquoise 90%, turquoise 100%);
+    }
+  </style>
+</head>
+<amp-base-carousel id="my-carousel">
+  <div class="red-gradient"></div>
+  <div class="blue-gradient"></div>
+  <div class="green-gradient"></div>
+</amp-base-carousel>
+<div class="buttons" style="margin-top: 8px;">
+  <button id='prev-button'>Go to previous slide</button>
+  <button id='next-button'>Go to next slide</button>
+  <button id='go-to-button'>Go to slide with green gradient</button>
+</div>
+
+<script>
+  (async () => {
+    const carousel = document.querySelector('#my-carousel');
+    await customElements.whenDefined('amp-base-carousel');
+    const api = await carousel.getApi();
+
+    // programatically advance to next slide
+    api.next();
+
+    // set up button actions
+    document.querySelector('#prev-button').onclick = () => api.prev();
+    document.querySelector('#next-button').onclick = () => api.next();
+    document.querySelector('#go-to-button').onclick = () => api.goToSlide(2);
+  })();
+</script>
+```
+
+#### Interactivity and API usage
+
+Bento enabled components in standalone use are highly interactive through their API. In Bento standalone use, the element's API replaces AMP Actions and events and [`amp-bind`](https://amp.dev/documentation/components/amp-bind/?format=websites).
+
+The `amp-base-carousel` component API is accessible by including the following script tag in your document:
+
+```
+await customElements.whenDefined('amp-base-carousel');
+const api = await carousel.getApi();
+```
+
+##### Actions
+
+The `amp-base-carousel` API allows you to perform the following actions:
+
+**next()**
+Moves the carousel forwards by `advance-count` slides.
+
+```
+api.next();
+```
+
+**prev()**
+Moves the carousel backwards by `advance-count` slides.
+
+```
+api.prev();
+```
+
+**goToSlide(index: number)**
+Moves the carousel to the slide specified by the `index` argument.
+
+```
+api.goToSlide(0); // Advance to first slide.
+api.goToSlide(length - 1); // Advance to last slide.
+```
+
```

Comment:
Do they need to include `index:` in the argument? Or just the index number?

Reply:
Just the index. The `index: number` is to give the parameter a type, but let me specify this inline instead so it's less confusing, something like this:
```suggestion
**goToSlide(index)**
Args:
  - `index`: number [required]
Moves the carousel to the slide specified by the `index` argument.

\```
api.goToSlide(0); // Advance to first slide.
api.goToSlide(length - 1); // Advance to last slide.
\```

```

Reply:
TypeScript-like notation here I think is the best in the long term. We are also likely to publish TypeScript type declarations (based on our `x.type.js`) in the future too.

Reply:
@dvoytenko are you suggesting keep as is?